pkg/agent/qrm-plugins/memory/dynamicpolicy/oom/calculator.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oom
+
+import (
+	"fmt"
+
+	core "k8s.io/api/core/v1"
+
+	"github.com/kubewharf/katalyst-core/pkg/config/generic"
+	"github.com/kubewharf/katalyst-core/pkg/util/qos"
+)
+
+type OOMPriorityCalculator func(qosConfig *generic.QoSConfiguration, pod *core.Pod, previousPriority int) (int, error)
+
+var oomPriorityCalculators = []OOMPriorityCalculator{
+	nativeOOMPriorityCalculator,
+}
+
+func AppendOOMPriorityCalculator(calculator OOMPriorityCalculator) {
+	oomPriorityCalculators = append(oomPriorityCalculators, calculator)
+}
+
+func GetOOMPriority(qosConfig *generic.QoSConfiguration, pod *core.Pod) (oomPriority int, err error) {
+	for _, calculator := range oomPriorityCalculators {
+		oomPriority, err = calculator(qosConfig, pod, oomPriority)
+		if err != nil {
+			return 0, fmt.Errorf("calculator %T failed: %w", calculator, err)
+		}
+	}
+	return
+}
+
+func nativeOOMPriorityCalculator(qosConfig *generic.QoSConfiguration, pod *core.Pod, _ int) (int, error) {
+	qosLevel, err := qosConfig.GetQoSLevelForPod(pod)
+	if err != nil {
+		return 0, err
+	}
+
+	userSpecifiedOOMPriority, _ := qos.GetOOMPriority(qosConfig, pod)
```

Comment:
why not check invalid and return error here?

Reply:
Since we considered using the default value corresponding to the QoS level when the user inputs an invalid format. If an error is returned here, this situation will be ignored

Reply:
after discussing with other maintainers, we idendity that user inputs an invalid format and log error instead of using the default value corresponding to the QoS level.

Reply:
done